src/test/java/org/folio/rest/api/RequestUpdateTriggerTest.java

```diff
@@ -62,21 +66,30 @@ void updateRequestTriggerTest(String oldStatus,
                                 boolean awaitingPickupRequestClosedDatePresent)
     throws InterruptedException, ExecutionException, TimeoutException {
 
-    CompletableFuture<Request> future = new CompletableFuture<>();
+    CompletableFuture<JsonObject> future = new CompletableFuture<>();
 
     String id = "3a57dc83-e70d-404b-b1f1-442b88760331";
 
     Request request = new Request()
       .withStatus(fromValue(oldStatus));
 
+    DateTime requestUpdatedDate = DateTime.now();
+
     saveRequest(id, request)
       .compose(v -> updateRequest(id, request.withStatus(fromValue(newStatus))))
       .compose(v -> getRequest(id))
       .setHandler(updatedRequest -> future.complete(updatedRequest.result()));
 
-    Request updatedRequest = future.get(5, TimeUnit.SECONDS);
+    JsonObject updatedRequest = future.get(5, TimeUnit.SECONDS);
+
 
-    assertThat(updatedRequest.getAwaitingPickupRequestClosedDate() != null, is(awaitingPickupRequestClosedDatePresent));
+    if (awaitingPickupRequestClosedDatePresent) {
```

Comment:
Rather than introduce an if statement in the tests, an alternative is to assert for the presence of the property and then if the value is similar to expected.

If the first assert fails, the test will stop, there is no need for conditional execution.

Reply:
Right, but the property can be not presented and it is normal scenario. The property presence is a parameter of parameterized test defined in Csv source https://github.com/folio-org/mod-circulation-storage/pull/164/files#diff-aef83ff3f59a523d89c77c8b22f052c8R58

So if we are not expecting this property presented than we don't need to perform the check if the value is similar to expected.

Reply:
If we are not expecting the property to be present, we should specifically check for that.

Oh, personally I would split the scenarios where the property is expected to be present and where it is not.

I find combining them together and using conditionals in the test makes it harder to understand the behaviour and increases the chances that the tests are brittle or unreliable.

Reply:
Agree that conditional in the test looks bad.

Initially I decided to use such approach (parameterized test with csv source) to keep all scenarios in one place. Where each row is test's parameters.

For example:
`"Open - Awaiting pickup | Closed - Pickup expired | true"` means that the property is present after status transition from `Open - Awaiting pickup` to `Closed - Pickup expired`.

I can get rid of `awaitingPickupRequestClosedDatePresent` test parameter  and split the test into two ones, the first for scenarios when the property must be present and the second one where not.

Reply:
Thank you

Reply:
Split the test and added more cases for scenario when the property should not be presented.